external-import/mandiant/src/connector/base.py

```diff
@@ -529,112 +513,219 @@ def compute_start_structure(self, start_date):
+                f"State key {state_key} not found in {collection_name} collection"
+            ) from err
 
-            if collection in collection_with_start_epoch:
-                first_run = (
-                    True
-                    if state[collection][STATE_START]
-                    == Timestamp.from_iso(import_start_date).iso_format
-                    else False
+    def set_state_value(self, collection_name: str, state_key: str, value: Any):
+        """Using this method to set the value of a specific key in the state collection.
+
+        See Also:
+            get_state_value
+        """
+        try:
+            state = self.helper.get_state()
+            state[collection_name][state_key] = value
+            self.helper.set_state(state)
+        except (KeyError, TypeError) as err:
+            raise StateError(
+                f"State key {state_key} not found in {collection_name} collection"
+            ) from err
+
+    def process_message(self):
+        self._init_state()
+        for collection in self.mandiant_collections:
+            try:
+                # Handle interval config
+                work_id = None
+                date_now_value = Timestamp.now().value
+                collection_interval = getattr(self, f"mandiant_{collection}_interval")
+
+                last_run_value = Timestamp.from_iso(
+                    self.get_state_value(
+                        collection_name=collection, state_key=STATE_LAST_RUN
+                    )
+                ).value
+
+                # API types related to simple offset
+                collection_with_offset = ["malwares", "actors", "campaigns"]
+                # Start and End, Offset
+                start_offset = self.get_state_value(
+                    collection_name=collection, state_key=STATE_OFFSET
                 )
-            else:
-                first_run = True if start_offset == 0 else False
+                end_offset = start_offset + OFFSET_PAGINATION
+
+                # API types related to start_epoch
+                collection_with_start_epoch = [
+                    "reports",
+                    "vulnerabilities",
+                    "indicators",
+                ]
+                # Start and End, Timestamp short format
+                start_date = Timestamp.from_iso(
+                    self.get_state_value(
+                        collection_name=collection, state_key=STATE_START
+                    )
+                )
+                start_short_format = start_date.short_format
 
-            """
-            We check that after each API call the collection respects the interval, 
-            either the default or the one specified in the config.
-            If it does not, we terminate the job and move on to the next collection.
-            """
+                # If no end date, put the proper period using delta
+                if (
+                    self.get_state_value(
```

Comment:
use of safe get_state_value responsible for handling the Error rather than direct key access